authorize_error_test.go

```diff
@@ -70,12 +70,12 @@ func TestWriteAuthorizeError(t *testing.T) {
 			err: ErrInvalidGrant,
 			mock: func(rw *MockResponseWriter, req *MockAuthorizeRequester) {
 				req.EXPECT().IsRedirectURIValid().Return(false)
-				rw.EXPECT().Header().Return(header)
+				rw.EXPECT().Header().AnyTimes().Return(header)
 				rw.EXPECT().WriteHeader(http.StatusBadRequest)
 				rw.EXPECT().Write(gomock.Any())
 			},
 			checkHeader: func(t *testing.T, k int) {
-				assert.Equal(t, "application/json", header.Get("Content-Type"))
+				assert.Equal(t, "application/json;charset=UTF-8", header.Get("Content-Type"))
```

Comment:
Note that these changes were needed to have tests passing. They are not semantically related to the rest of this PR.

Reply:
I think these changes are already done in https://github.com/ory/fosite/pull/466/files

I suggest you remove them from this PR for now.

Reply:
Done.